src/plugins/ucx/ucx_backend.cpp

```diff
@@ -244,7 +246,21 @@ class nixlUcxBackendH : public nixlBackendReqH {
     nixlUcxIntReq head;
     nixlUcxWorker* uw;
 
+    // Notification to be sent after completion of all requests
+    std::optional<std::pair<std::string,std::string>> notif;
+
 public:
+    void notif_schedule(const std::string& agent,
+                        const std::string& message) {
+        notif = std::make_pair(agent, message);
+    }
+
+    std::pair<const std::string&,const std::string&> notif_args() const {
+        return {notif.value().first, notif.value().second};
```

Comment:
it will throw an exception if notif is not set, maybe return special value in this case instead?

Reply:
value_or can be used here